src/core/eme/session_events_listener.ts

```diff
@@ -138,9 +138,8 @@ export default function SessionEventsListener(
         const getLicenseTimeout = isNullOrUndefined(getLicenseConfig.timeout) ?
           10 * 1000 :
           getLicenseConfig.timeout;
-        return (castToObservable(getLicense) as Observable<Uint8Array|null>)
-          .pipe(map(license => license !== null ? new Uint8Array(license) : null),
-                getLicenseTimeout >= 0 ? timeout(getLicenseTimeout) :
+        return (castToObservable(getLicense) as Observable<BufferSource|null>)
```

Comment:
Why casting is necessary at all here? What is inferred if we do not?

Reply:
After doing some research, I think Typescript can't infer the type cause of the function `castToObservable` and his type signature.

The current signature is this:

`src/utils/cast_to_observable.ts`
```
function castToObservable<T>(value? : T) : Observable<T>;
```

So in our case, the typescript infers the type with:
`castToObservable<ArrayBufferView | ArrayBuffer | Promise<BufferSource | null> | null>(...)`
And so, the same type will be returned while we have to return an `Observable<BufferSource | null>`

I managed to get it to work by editing the signature but it would mean refactoring elsewhere:
```
function castToObservable<T, J>(value? : T) : Observable<J>;
```

`castToObservable<ArrayBufferView | ArrayBuffer | Promise<BufferSource | null> | null, ILicense | null>(getLicense)`